source/credentials_provider_sts.c

```diff
@@ -780,29 +794,28 @@ struct aws_credentials_provider *aws_credentials_provider_new_sts(
 
     impl->role_session_name =
         aws_string_new_from_array(allocator, options->session_name.ptr, options->session_name.len);
-
-    if (!impl->role_session_name) {
-        goto on_done;
-    }
-
```

Comment:
We should throw a proper error if these strings are required instead of creating a string with 0 len.